React/CoreModules/RCTDevSettings.mm

```diff
@@ -439,13 +440,14 @@ - (void)setupHotModuleReloadClientIfApplicableForURL:(NSURL *)bundleURL
     NSString *const path = [bundleURL.path substringFromIndex:1]; // Strip initial slash.
     NSString *const host = bundleURL.host;
     NSNumber *const port = bundleURL.port;
+    // TODO(macOS ISS#2323203) - we could perhaps infer the platform from the bundleURL's query parameters, instead of hardcoding
```

Comment:
The `bundleURL` has a query parameter like `platform=macos`. If that parameter is always expected to be available, then we could reduce our forked surface by using that query parameter upstream.